web/json_value.go

```diff
@@ -82,14 +83,28 @@ func convertJSONAttributeValue(attribute *framework.AttributeConfig, value any,
 		return boolValue, nil
 
 	case framework.AttributeTypeDateTime:
-		v, ok := value.(string)
-		if !ok {
-			return nil, fmt.Errorf("attribute %s cannot be parsed into a string date-time value", attribute.ExternalId)
+		var dateTimeStr string
+		switch v := value.(type) {
+		case string:
+			dateTimeStr = v
+		case float64:
+			// make sure the value is in int64 range
+			if v > float64(math.MaxInt64) || v < float64(math.MinInt64) {
+				return nil, fmt.Errorf("attribute %s cannot be parsed into a date-time value as the value is out of the valid range", attribute.ExternalId)
+			}
+			// make sure float only has 0 decimals (e.g. 123.00000)
+			if float64(int(v)) != v {
```

Comment:
Let's keep the conversions consistent
```suggestion
			if float64(int64(v)) != v {
```